Fix [v4.4.0] TypeScript reports errors

- Markuplint version: `4.4.0`
- TypeScript version: `5.4.3`
- Node.js version: `v20.11.1`
- OS: Ubuntu 22.04.4 LTS

## Describe the bug

TypeScript reports errors in Markuplint. I think the problem is that some imports are prefixed with `"packages/"`.

## Code Example

- `package.json`

  ```JSON
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "markuplint": "4.4.0",
      "typescript": "5.4.3"
    }
  }
  ```

- `index.ts`

  ```TypeScript
  import { MLEngine } from "markuplint";

  console.log(MLEngine);
  ```

## Steps To Reproduce

1. `npm install`
2. `npx tsc --lib esnext,dom --module nodenext index.ts`

## The current behavior

```
node_modules/@markuplint/ml-core/lib/test/index.d.ts:13:110 - error TS2307: Cannot find module 'packages/@markuplint/ml-core/src/ml-dom/node/node.js' or its corresponding type declarations.

13 export declare function createTestNodeList(sourceCode: string, options?: CreateTestOptions): readonly import("packages/@markuplint/ml-core/src/ml-dom/node/node.js").MLNode<any, any, import("@markuplint/ml-ast").MLASTNode>[];
                                                                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/test/index.d.ts:14:111 - error TS2307: Cannot find module 'packages/@markuplint/ml-core/src/ml-dom/token/token.js' or its corresponding type declarations.

14 export declare function createTestTokenList(sourceCode: string, options?: CreateTestOptions): readonly import("packages/@markuplint/ml-core/src/ml-dom/token/token.js").MLToken<import("@markuplint/ml-ast").MLASTToken>[];
                                                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/test/index.d.ts:15:100 - error TS2307: Cannot find module 'packages/@markuplint/ml-core/src/ml-dom/node/element.js' or its corresponding type declarations.

15 export declare function createTestElement(sourceCode: string, options?: CreateTestOptions): import("packages/@markuplint/ml-core/src/ml-dom/node/element.js").MLElement<any, any>;
                                                                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-spec/lib/dom-traverse/get-implicit-role.d.ts:3:105 - error TS2307: Cannot find module 'packages/@markuplint/ml-spec/src/index.js' or its corresponding type declarations.

3 export declare function getImplicitRoleName(el: Element, version: ARIAVersion, specs: MLMLSpec): import("packages/@markuplint/ml-spec/src/index.js").ImplicitRole;
                                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/markuplint/lib/api/v1.d.ts:51:20 - error TS2307: Cannot find module 'packages/markuplint/src/types.js' or its corresponding type declarations.

51 }): Promise<import("packages/markuplint/src/types.js").MLResultInfo[]>;
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 5 errors in 3 files.

Errors  Files
     3  node_modules/@markuplint/ml-core/lib/test/index.d.ts:13
     1  node_modules/@markuplint/ml-spec/lib/dom-traverse/get-implicit-role.d.ts:3
     1  node_modules/markuplint/lib/api/v1.d.ts:51
```

## The expected behavior

No errors.

--- a/packages/@markuplint/ml-core/src/test/index.ts
+++ b/packages/@markuplint/ml-core/src/test/index.ts
@@ -1,4 +1,7 @@
-import type { MLParser } from '@markuplint/ml-ast';
+import type { MLElement } from '../ml-dom/node/element.js';
+import type { MLNode } from '../ml-dom/node/node.js';
+import type { MLToken } from '../ml-dom/token/token.js';
+import type { MLASTNode, MLASTToken, MLParser } from '@markuplint/ml-ast';
 import type { Config, PlainData, RuleConfigValue } from '@markuplint/ml-config';
 import type { ExtendedSpec, MLMLSpec } from '@markuplint/ml-spec';
 
@@ -6,7 +9,7 @@ import { parser } from '@markuplint/html-parser';
 import spec from '@markuplint/html-spec';
 
 import { convertRuleset } from '../convert-ruleset.js';
-import { Document } from '../ml-dom/index.js';
+import { MLDocument } from '../ml-dom/node/document.js';
 
 export type CreateTestOptions = {
 	readonly config?: Config;
@@ -17,28 +20,31 @@ export type CreateTestOptions = {
 export function createTestDocument<T extends RuleConfigValue = any, O extends PlainData = any>(
 	sourceCode: string,
 	options?: CreateTestOptions,
-) {
+): MLDocument<T, O> {
 	const ast = options?.parser
 		? 'parser' in options.parser
 			? options.parser.parser.parse(sourceCode, options.config?.parserOptions)
 			: options.parser.parse(sourceCode, options.config?.parserOptions)
 		: parser.parse(sourceCode, options?.config?.parserOptions);
 	const ruleset = convertRuleset(options?.config);
-	const document = new Document<T, O>(ast, ruleset, [options?.specs ?? ({} as any), {}]);
+	const document = new MLDocument<T, O>(ast, ruleset, [options?.specs ?? ({} as any), {}]);
 	return document;
 }
 
-export function createTestNodeList(sourceCode: string, options?: CreateTestOptions) {
+export function createTestNodeList(
+	sourceCode: string,
+	options?: CreateTestOptions,
+): readonly MLNode<any, any, MLASTNode>[] {
 	const document = createTestDocument(sourceCode, options);
 	return document.nodeList;
 }
 
-export function createTestTokenList(sourceCode: string, options?: CreateTestOptions) {
+export function createTestTokenList(sourceCode: string, options?: CreateTestOptions): readonly MLToken<MLASTToken>[] {
 	const document = createTestDocument(sourceCode, options);
 	return document.getTokenList();
 }
 
-export function createTestElement(sourceCode: string, options?: CreateTestOptions) {
+export function createTestElement(sourceCode: string, options?: CreateTestOptions): MLElement<any, any> {
 	const document = createTestDocument(sourceCode, options);
 	const el = document.nodeList[0];
 	if (el && el.is(el.ELEMENT_NODE)) {

--- a/packages/@markuplint/ml-spec/src/dom-traverse/get-implicit-role.ts
+++ b/packages/@markuplint/ml-spec/src/dom-traverse/get-implicit-role.ts
@@ -1,3 +1,4 @@
+import type { ImplicitRole } from '../types/aria.js';
 import type { ARIAVersion, ComputedRole, MLMLSpec } from '../types/index.js';
 
 import { getImplicitRole as _getImplicitRole } from '../specs/get-implicit-role.js';
@@ -41,6 +42,6 @@ export function getImplicitRoleName(
 	el: Element,
 	version: ARIAVersion,
 	specs: MLMLSpec,
-) {
+): ImplicitRole {
 	return _getImplicitRole(specs, el.localName, el.namespaceURI, version, el.matches.bind(el));
 }

--- a/packages/markuplint/src/api/v1.ts
+++ b/packages/markuplint/src/api/v1.ts
@@ -1,3 +1,4 @@
+import type { MLResultInfo } from '../types.js';
 import type { Config, PlainData, RuleConfigValue } from '@markuplint/ml-config';
 import type { MLRule } from '@markuplint/ml-core';
 
@@ -62,7 +63,7 @@ export async function lint_v1(options: {
 	 * Locale
 	 */
 	readonly locale?: string;
-}) {
+}): Promise<MLResultInfo[]> {
 	const filePathList = toNoEmptyStringArrayFromStringOrArray(options.files);
 	const codes = toNoEmptyStringArrayFromStringOrArray(options.sourceCodes);
 

--- a/test/isolated-env/ts/tsconfig.json
+++ b/test/isolated-env/ts/tsconfig.json
@@ -2,7 +2,6 @@
 	"compilerOptions": {
 		"target": "ESNext",
 		"module": "NodeNext",
-		"moduleResolution": "NodeNext",
-		"skipLibCheck": true
+		"moduleResolution": "NodeNext"
 	}
 }

